closes #2085: publish v2 quarkus-common jar to the maven repository

ivansenic opened the pull request:
**What this PR does**:
Publishes `svg2-quarkus-common` to the Maven repository:

* uses similar setup as the base pom
* however, with the newest plugin versions
* I had to do some gymnastics to enable only `svg2-quarkus-common` in the deploy profile

**Which issue(s) this PR fixes**:
Fixes #2085

jeffreyscarpenter commented:
Just wondering if publishing the service Jars is such a bad thing, especially if we have to put in so much effort to avoid publishing them?

tatu-at-datastax commented on apis/pom.xml:
```diff
@@ -162,5 +174,111 @@
+      </build>
+    </profile>
   </profiles>
+  <licenses>
```
Maybe move "developers", "licenses" and "scm" up towards beginning of file.

ivansenic replied:
where exactly would this fit?

tatu-at-datastax replied:
Earlier after artifact and group ids, description; before dependencies and properties is typical.

But whatever, it's fine.

tatu-at-datastax reviewed (approved):
LGTM

tatu-at-datastax commented:
Looks like "Unit tests" fails with:

```
java.lang.RuntimeException: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for io.stargate:sgv2-docsapi::jar:2.0.0-BETA-3-SNAPSHOTnull
```

so something missing/wrong in `pom.xml` or CI.

ivansenic commented:
> Just wondering if publishing the service Jars is such a bad thing, especially if we have to put in so much effort to avoid publishing them?

Who would ever use this?

ivansenic commented:
> Looks like "Unit tests" fails with:
> 
> ```
> java.lang.RuntimeException: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for io.stargate:sgv2-docsapi::jar:2.0.0-BETA-3-SNAPSHOTnull
> ```
> 
> so something missing/wrong in `pom.xml` or CI.

Don't get this, I can not reproduce locally.. Doing exactly the same thing as in the CI.. I do assume it's something with the modules, or some caching issues maybe..

ivansenic commented:
@tatu-at-datastax seems that the modules in profiles was not working, so I reverted and added a `-pl sgv2-quarkus-common` to the github workflow.. Can you please re-check again?

tatu-at-datastax commented:
> @tatu-at-datastax seems that the modules in profiles was not working, so I reverted and added a `-pl sgv2-quarkus-common` to the github workflow.. Can you please re-check again?

Was just looking at CI, seems to pass now.
